tracer/src/Datadog.Trace/IAST/IastModule.cs

```diff
@@ -64,11 +65,28 @@ public IastModule()
 
         if (!iast.Settings.DeduplicationEnabled || HashBasedDeduplication.Instance.Add(vulnerability))
         {
-            // The VulnerabilityBatch class is not very useful right now, but we will need it when handling requests
+            return AddVulnerability(tracer, integrationId, operationName, vulnerability);
+        }
+
+        return null;
+    }
+
+    private static Scope? AddVulnerability(Tracer tracer, IntegrationId integrationId, string operationName, Vulnerability vulnerability)
+    {
+        var iActiveScope = DistributedTracer.Instance.GetActiveScope() ?? tracer.TracerManager?.ScopeManager.Active;
```

Comment:
`Tracer.Active` already handles this:
```suggestion
        var iActiveScope = tracer.Active;
```